### PR TITLE
MDEV-34647 : 'INSERT...SELECT' on MyISAM table suddenly replicated by…

### PR DESCRIPTION
**Pull request by janlindstrom**
… Galera

Replication of MyISAM and Aria DML is experimental and best effort only. Earlier change make INSERT SELECT on both MyISAM and Aria to replicate using TOI and STATEMENT replication. Replication should happen only if user has set needed wsrep_mode setting.

### CODE REVIEW

File: sql/sql_parse.cc [hunk 1]
```diff
@@ -4719,7 +4719,15 @@ mysql_execute_command(THD *thd, bool is_called_from_prepared_stmt)
 
         // For !InnoDB we start TOI if it is not yet started and hope for the best
         if (!is_innodb && !wsrep_toi)
-          WSREP_TO_ISOLATION_BEGIN(first_table->db.str, first_table->table_name.str, NULL);
+        {
+          const legacy_db_type db_type= first_table->table->file->partition_ht()->db_type;
+          bool replicate= ((db_type == DB_TYPE_MYISAM && wsrep_check_mode(WSREP_MODE_REPLICATE_MYISAM)) ||
+                           (db_type == DB_TYPE_ARIA && wsrep_check_mode(WSREP_MODE_REPLICATE_ARIA)));
+
+          /* Currenctly we support TOI for MyISAM and Aria only. */
```
Comment (sciascid) [inline sql/sql_parse.cc:4727]: Typo: "Currenctly" should be "Currently"

File: sql/sql_parse.cc [hunk 2]
```diff
@@ -4719,7 +4719,15 @@ mysql_execute_command(THD *thd, bool is_called_from_prepared_stmt)
 
         // For !InnoDB we start TOI if it is not yet started and hope for the best
         if (!is_innodb && !wsrep_toi)
-          WSREP_TO_ISOLATION_BEGIN(first_table->db.str, first_table->table_name.str, NULL);
+        {
+          const legacy_db_type db_type= first_table->table->file->partition_ht()->db_type;
+          bool replicate= ((db_type == DB_TYPE_MYISAM && wsrep_check_mode(WSREP_MODE_REPLICATE_MYISAM)) ||
```
Comment (sciascid) [inline sql/sql_parse.cc:4724]: Variable `replicate` can be made `const`

**Review by sciascid — CHANGES_REQUESTED** [review]
Requested minor changes, see comments above. Looks good otherwise.

File: sql/sql_parse.cc (hunk 2 above)
Comment (janlindstrom) [reply sql/sql_parse.cc]: Fixed

**Pull request closed without merging**